draft-jhoyla-req-mtls-flag.md

```diff
@@ -32,48 +32,99 @@ informative:
 --- abstract
 
 Normally in TLS there is no way for the client to signal to the server that it
-has been configured with a certificate suitable for mTLS. This document defines
-a TLS Flag {{!I-D.ietf-tls-tlsflags}} that enables clients to provide this hint.
+supports client authentication and will handle a `CertificateRequest`
+gracefully. This document defines a TLS Flag that enables clients to provide
+this hint.
 
 
 --- middle
 
 # Introduction
 
-This document specifies a TLS Flag that indicates to the server that the client
-supports mTLS. Sometimes a server does not want to negotiate mTLS with every
-client, but might wish to authenticate a subset of them. In TLS 1.3 this may be
-done with post-handshake auth, however this adds an extra round-trip, and
-requires negotiation at the application layer. A client sending the request
-mTLS flag in the ClientHello allows the server to request authentication during
-the initial handshake only when it receives a hint the client supports it. This
-enables a number of use cases, for example allowing bots to authenticate
-themselves when mixed in with general traffic.
+This document specifies a TLS Flag {{!I-D.ietf-tls-tlsflags}} that allows a
+client to attempt to negotiate mutually authenticated TLS (in at least some
+cases). Sometimes a server does not want to authenticate every client, but
+might wish to authenticate a subset of them. In TLS 1.3 this may be done with
+post-handshake authentication, however this adds an extra round trip, and
+requires negotiation at the application layer.
+
+The behaviour specified in {{?RFC8446}} for a client that receives a
+`CertificateRequest` that it cannot satisfy is to send an empty Certificate
+message followed by a `Finished` message. However in practice many clients
+simply terminate the connection in anticipation of a "certificate_required"
+alert.
+
+This behaviour makes it difficult for servers to send `CertificateRequest`
+messages in the wild. A client sending the `request_client_auth` flag in the
+`ClientHello` allows the server to request authentication during the initial
+handshake only when it receives a hint the client will handle the request
+without terminating the connection, and, if unable to authenticate, by sending
+an empty certificate message, per {{?RFC8446, Section 4.4.2}}. This enables a
+number of use cases, for example allowing bots to authenticate themselves when
+mixed in with general traffic.
+
+This flag is intended to be used by clients sending automated traffic, not
+those that are directly operated by (human) users. Clients that set this flag
+SHOULD only do so when a client certificate is directly available to them, and
```

Comment:
A bit nit-picky, but: what does "directly available" mean? A bot might still need to select a cert based on the contents of the CR.

Reply:
The goal is to make it clear that you shouldn't set this flag if you aren't certain you have at least one certificate available. You don't want the client to start trying to enumerate keys on a smartcard or prompting the user for a PEM file. This is only supposed to be used if you already know what certificate(s) you have available.